src/webgpu/api/operation/rendering/stencil.spec.ts

```diff
@@ -147,31 +153,73 @@ class StencilTest extends GPUTest {
     };
 
     const encoder = this.device.createCommandEncoder();
-    const pass = encoder.beginRenderPass({
-      colorAttachments: [
-        {
-          view: renderTarget.createView(),
-          storeOp: 'store',
-          loadOp: 'load',
+    let pass: GPURenderPassEncoder;
+    if (isSingleEncoderMultiplePass) {
+      // Clear the depthTextureView at the beginning.
+      pass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: renderTarget.createView(),
+            storeOp: 'store',
+            loadOp: 'clear',
+          },
+        ],
+        depthStencilAttachment: {
+          view: depthTexture.createView(),
+          depthLoadOp: 'clear',
+          depthStoreOp: 'store',
+          stencilLoadOp: 'clear',
+          stencilStoreOp: 'store',
         },
-      ],
-      depthStencilAttachment,
-    });
+      });
+      pass.end();
+    } else {
+      pass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: renderTarget.createView(),
+            storeOp: 'store',
+            loadOp: 'load',
+          },
+        ],
+        depthStencilAttachment,
+      });
+    }
 
     // Draw a triangle with the given stencil reference and the comparison function.
     // The color will be kGreenStencilColor if the stencil test passes, and kBaseColor if not.
     for (const test of testStates) {
+      if (isSingleEncoderMultiplePass) {
+        pass = encoder.beginRenderPass({
+          colorAttachments: [
+            {
+              view: renderTarget.createView(),
+              storeOp: 'store',
+              loadOp: 'load',
+            },
+          ],
+          depthStencilAttachment,
+        });
+      }
       const testPipeline = this.createRenderPipelineForTest(test.state);
       pass.setPipeline(testPipeline);
-      pass.setStencilReference(test.stencil);
+      if (test.setStencilReference || test.setStencilReference === undefined) {
```

Comment:
I think this should work to "default to true"
```suggestion
      if (test.setStencilReference ?? true) {
```

Reply:
Thanks, it's simpler.